File: LeetCode/remove_duplicates_from_sorted_array.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        if(!nums.size())
            return 0;
        int aux = 0, prev = -1;
        nums[aux++] = nums[0];
        for(int i = 1; i < nums.size(); i++) {
           if(nums[i] != nums[aux-1]) {
               nums[aux++] = nums[i];
           }
        }
        return aux;
    }
};

File: LeetCode/remove_duplicates_from_sorted_array.go
func removeDuplicates(nums []int) int {
    pos := 1
    prev := nums[0]

    for i := 1; i < len(nums); i++ {
        if prev != nums[i] {
            nums[pos] = nums[i]
            pos++
            prev = nums[i]
        }
    }

    return pos
}
